File: app/java/src/main/java/com/example/production/domain/model/inventory/Stocktaking.java
package com.example.production.domain.model.inventory;

import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 棚卸データエンティティ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Stocktaking {
    private Long id;
    private String stocktakingNumber;
    private String locationCode;
    private LocalDate stocktakingDate;
    private StocktakingStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private List<StocktakingDetail> details;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Inventory/Stocktaking.cs
namespace ProductionManagement.Domain.Models.Inventory;

/// <summary>
/// 棚卸データ
/// </summary>
public class Stocktaking
{
    private string _statusValue = "発行済";

    public long? Id { get; init; }
    public required string StocktakingNumber { get; init; }
    public required string LocationCode { get; init; }
    public required DateOnly StocktakingDate { get; init; }

    /// <summary>
    /// ステータス（データベース格納用の日本語文字列）
    /// </summary>
    public string StatusValue
    {
        get => _statusValue;
        set => _statusValue = value;
    }

    /// <summary>
    /// ステータス（ドメインロジック用の Enum）
    /// </summary>
    public StocktakingStatus Status
    {
        get => StocktakingStatusExtensions.FromDisplayName(_statusValue);
        init => _statusValue = value.ToDisplayName();
    }

    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }

    public IReadOnlyList<StocktakingDetail>? Details { get; set; }
}
